inst/template/mode.R.template

```diff
@@ -35,5 +35,10 @@
 
     solve = function(end_time) {
       mode_{{name}}_solve(private$ptr_, end_time)
+    },
+
+    update_state = function(time) {
```

Comment:
Following from the comment in the testing, I wonder if it's worth adding the full interface here:

```
    update_state = function(pars = NULL, state = NULL, time = NULL,
                            set_initial_state = NULL) {
```

and erroring if anything we don't support is given? Alternatively just take treat these essentially as keyword args (rather than positional) for now

Reply:
ok, have added this but not doing anything with the other args for now